twilight-http/src/request/guild/auto_moderation/create_auto_moderation_rule.rs

```diff
@@ -229,51 +244,104 @@ impl<'a> CreateAutoModerationRule<'a> {
+    ///
+    /// Returns [`AutoModerationMetadataKeywordFilter`] if the `keyword_filter`
+    /// field is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataKeywordFilterItem`] if a `keyword_filter`
+    /// item is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataAllowList`] if the `allow_list` field is
+    /// invalid.
+    ///
+    /// Returns [`AutoModerationMetadataAllowListItem`] if an `allow_list` item
+    /// is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataRegexPatterns`] if the `regex_patterns`
+    /// field is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataRegexPatternsItem`] if a `regex_patterns`
+    /// item is invalid.
     ///
     /// [`Keyword`]: AutoModerationTriggerType::Keyword
     /// [Discord Docs/Keyword Matching Strategies]: https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies
     /// [Discord Docs/Trigger Metadata]: https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-trigger-metadata
+    /// [`AutoModerationMetadataKeywordFilter`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataKeywordFilter
+    /// [`AutoModerationMetadataKeywordFilterItem`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataKeywordFilterItem
+    /// [`AutoModerationMetadataAllowList`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataAllowList
+    /// [`AutoModerationMetadataAllowListItem`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataAllowListItem
+    /// [`AutoModerationMetadataRegexPatterns`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataRegexPatterns
+    /// [`AutoModerationMetadataRegexPatternsItem`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataRegexPatternsItem
     pub fn with_keyword(
         mut self,
         keyword_filter: &'a [&'a str],
+        regex_patterns: &'a [&'a str],
+        allow_list: &'a [&'a str],
     ) -> ResponseFuture<AutoModerationRule> {
-        self.fields = self.fields.map(|mut fields| {
+        self.fields = self.fields.and_then(|mut fields| {
+            validate_auto_moderation_metadata_keyword_allow_list(allow_list)?;
+            validate_auto_moderation_metadata_keyword_filter(keyword_filter)?;
+            validate_auto_moderation_metadata_regex_patterns(regex_patterns)?;
             fields.trigger_metadata = Some(CreateAutoModerationRuleFieldsTriggerMetadata {
-                allow_list: None,
+                allow_list: Some(allow_list),
                 keyword_filter: Some(keyword_filter),
                 presets: None,
                 mention_total_limit: None,
+                regex_patterns: Some(regex_patterns),
             });
 
             fields.trigger_type = Some(AutoModerationTriggerType::Keyword);
 
-            fields
+            Ok(fields)
         });
 
         self.exec()
```

Comment:
`exec` method should be deprecated, cant mark the actual method since it's outside the pr changes, but i don't think that change alone should have its own pr